### DISCUSSION TITLE
Problem with freezing script that imports rasterio package

### DISCUSSION THREAD

**Discussion by mkleinha** [marcelotduarte/cx_Freeze#2454] [Q&A] [answered]
I am attempting to build an executable version of a python interface for visualizing some geospatial data. One of the things this program needs to do is read the value of a GeoTIFF at coordinates entered by the user. I have been using the `rasterio` package to implement this functionality, but I am having trouble getting the frozen .exe to import the `rasterio` package. The simplest reproduceable version of this issue is a script with just one line:
```
import rasterio
```

I use the basic cxfreeze command:
```
cxfreeze --script rasterio_import.py
```
When I try to run the resulting executable, I get this error (I replaced my username with {USERNAME}):
```
Traceback (most recent call last):
  File "C:\Users\{USERNAME}\AppData\Local\Programs\Python\Python312\Lib\site-packages\cx_Freeze\initscripts\__startup__.py", line 141, in run
    module_init.run(name + "__main__")
  File "C:\Users\{USERNAME}\AppData\Local\Programs\Python\Python312\Lib\site-packages\cx_Freeze\initscripts\console.py", line 25, in run
    exec(code, main_globals)
File "rasterio_import.py", line 1, in <module>
    import rasterio
  File "C:\Users\{USERNAME}\AppData\Local\Programs\Python\Python312\Lib\site-packages\rasterio\__init__.py", line 41, in <module>
    from rasterio._version import gdal_version, get_geos_version, get_proj_version
ImportError: DLL load failed while importing _version: The specified module could not be found.
```
I have tried installing all the necessary packages using both pip and conda and have tried isolating things using virtual environments and conda environments. I have tried multiple versions of all the necessary packages and always get the same error. I’ve tried adding the following lines (after installing the associated packages):
```
from osgeo import gdal 
```
```
import gdal
```
```
import pygdal
```
before the rasterio import as was suggested [here](https://github.com/conda-forge/rasterio-feedstock/issues/250) and in other threads.

I have spent considerable time with the cx_freeze documentation. I tried digging into the rasterio code to figure things out, but I’m stumped. I’ve got so many different programs that use GDAL successfully on this machine, but this one can’t seem to find it. 

I am using Windows 10
Python 3.12.1
rasterio 1.3.10
cx_Freeze 7.1.0

Please let me know if there is any other information that would be helpful in diagnosing this issue.

**Comment by marcelotduarte** [comment, marked as answer]
I made a PR #2455 to fix this.
Please test using pip and conda and give me feedback.
You can test the patch in the latest development build (it should be available within half an hour):
`pip install --force --no-cache --pre --extra-index-url https://marcelotduarte.github.io/packages/ cx_Freeze`

**Reply by mkleinha** [reply]
That seems to have worked. Thanks so much for resolving this so quickly.

**Reply by marcelotduarte** [reply]
[Release 7.1.1](https://github.com/marcelotduarte/cx_Freeze/releases/tag/7.1.1) is out!
[Documentation](https://cx-freeze.readthedocs.io)
